lib/middleware/proxy.js

```diff
@@ -22,23 +27,28 @@ module.exports = function(options) {
     return function(req, res, next) {
         if (req.url.indexOf('/__api__/proxy/') === 0 || req.url.indexOf('/proxy/') === 0) {
 
-            var url = decodeURIComponent(req.url.replace('/proxy/', '')),
-                targetHost = url.substring(0, url.indexOf('/', 8));
-
-            //Re-Build the location header, so browsers can follow them
+            var url = decodeURIComponent(req.url.replace('/proxy/', ''));
+			var targetHost = url.substring(0, url.indexOf('/', 8));
+			var path = url.substring(url.indexOf('/', 8), url.length);
+			
+			// Rewrite the path to not include the /proxy/ prefix
+			proxy.once('proxyReq', function (proxyReq, req, res) {
+				proxyReq.path = path;
+			});
+			
+			// Re-Build the location header, so browsers can follow them
             proxy.once('proxyRes', function (proxyRes, req, res) {
                 if (proxyRes.headers.location)
                     proxyRes.headers.location = 'http://' + req.headers.host + '/proxy/' + targetHost + proxyRes.headers.location;
             });
-
-            req.url = url;
-            delete req.headers.host;
-            proxy.web(req, res, { target: targetHost },
-                function error(err, req, res) {
+			
```

Comment:
Why did you remove:

```
req.url = url;
delete req.headers.host;
```
The removal of the host header solved issues for other users in the past.